File: arrays/javascript/singleNumber.js
var singleNumber = function(nums) {
    let freq = {}
    for (let i = 0; i < nums.length; i++) {
        if (!freq[nums[i]]) {
            freq[nums[i]] = 1
        } else {
            freq[nums[i]]++
        }
    }
    return Object.keys(freq).find(key => freq[key] === 1)
    
};

File: arrays/ruby/single_number.rb
def single_number(nums)
    #74%
    # hash = {}
    # nums.each do |x|
    #     if hash[x]
    #         hash[x]+=1
    #     else
    #         hash[x] = 1
    #     end
    # end
    # return hash.key(1)
    #97%
    2 * nums.uniq.sum - nums.sum
end
